### PR TITLE
fix(connection): add some log about datasource lock

### PR DESCRIPTION
**Pull request by ungreat**
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/datasource/SingleConnectionDataSource.java [hunk 1]
```diff
@@ -146,6 +147,7 @@ private Connection getConnectionProxy(@NonNull Connection connection) {
                     new Class[] {Connection.class},
                     new CloseIgnoreInvocationHandler(connection, this.lock));
         } catch (Exception e) {
+            log.info("get connection error unlock, hashcode=" + this.lock.hashCode());
```
Comment (yhilmare) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/datasource/SingleConnectionDataSource.java:150]: log.warn

File: server/odc-core/src/main/java/com/oceanbase/odc/core/datasource/SingleConnectionDataSource.java [hunk 2]
```diff
@@ -130,6 +130,7 @@ protected void prepareConnection(Connection con) throws SQLException {
 
     private boolean tryLock() {
         try {
+            log.info("get connection try lock, hashcode=" + this.lock.hashCode());
```
Comment (yhilmare) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/datasource/SingleConnectionDataSource.java:133]: we'd better get tryLock's result, if we obtain the lock and then we print this lock log~

**Review by yhilmare — APPROVED** [review]
lgtm

File: server/odc-core/src/main/java/com/oceanbase/odc/core/datasource/SingleConnectionDataSource.java (hunk 2 above)
Comment (ungreat) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/datasource/SingleConnectionDataSource.java]: That makes a lot of sense~

**Pull request merged**